packages/ap-frontend/docs/templatetags.rst

```diff
@@ -1,22 +1,665 @@
+
+So the following are all equivalent:
+
+.. code-block:: html+django
+
+    {% component "div" %}Hello World{% endcomponent %}
+
+    {% component %}
+        Hello world
+    {% endcomponent %}
+
+
+    {% component %}
+        Hello
+        world
+    {% endcomponent %}
+
+
+    {% component %}
+
+        Hello world
+    {% endcomponent %}
+
+Components can be nested:
+
+.. code-block:: html+django
+
+    {% component "components/Button" type="primary" %}
+        {% components "icons" "Menu" %}{% endcomponent %}
+        Open Menu
+    {% end_component %}
+
+and you can include HTML tags as children:
+
+.. code-block:: html+django
+
+    {% component "components/Button" type="primary" %}
+        <strong>Delete</strong> Item
+    {% end_component %}
+
+You can use ``as <variable name>`` to store in a variable in context that can then be passed to another tag:
+
+.. code-block:: html+django
+
+    {% component "icons" "Menu" as icon %}{% end_component %}
+    {% component "components/Button" type="primary" icon=icon %}Open Menu{% end_component %}
+
+All props must be JSON serializable. :class:`~alliance_platform.frontend.prop_handlers.ComponentProp` can be used to define
+how to serialize data, with a matching implementation in ``propTransformers.tsx`` to de-serialize it.
+
+For example :class:`~alliance_platform.frontend.prop_handlers.DateProp` handles serializing a python ``datetime`` and
+un-serializing it as a native JS ``Date`` on the frontend. See :class:`~alliance_platform.frontend.prop_handlers.ComponentProp`
+for documentation about adding your own complex props.
+
+Components are rendered using the ``renderComponent`` function in :data:`~alliance_platform.frontend.settings.AlliancePlatformFrontendSettingsType.REACT_RENDER_COMPONENT_FILE`. This can be modified as needed,
+for example if a new provider is required.
+
+.. note:: All props passed through are converted to camel case automatically (i.e. ``my_prop`` will become ``myProp``)
+
+Server Side Rendering (SSR)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Components will automatically be rendered on the server. See :ref:`ssr` for details about how this works.
+
+To opt out of SSR pass ``ssr:disabled=True`` to the component after the component name:
+
+.. code-block:: html+django
+
+    {% component 'components/Button.tsx' ssr:disabled=True %}...{% endcomponent %}
+
+Alternatively, you can disable SSR entirely by passing ``disable_ssr=True`` to :class:`~alliance_platform.frontend.bundler.vite.ViteBundler`.
+
+Options
+~~~~~~~
+
+Various options can be passed to the component tag. To differentiate from actual props to the component they are
+prefixed with `ssr:` for server side rendering options, `component:` for general component options, or `container:`
+for options relating to the container the component is rendered into.
+
+- ``ssr:disabled=True`` - if specified, no server side rendering will occur for this component
```

Comment:
If rendering on the client, will this automatically handle `<Suspense>` and displaying some sort of loading icon? If not, is that something we want to support (perhaps as an additional `ssr` argument), or should mention in the documentation?

Reply:
Suspense isn't applicable here as it's not loading it within a React tree. We could support an option for it, e.g. `ssr:placeholder=...` to allow rendering something.

I'll add a note about it when SSR is disabled

Reply:
https://github.com/AllianceSoftware/alliance-platform-py/issues/31 tracks this